connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -357,18 +358,16 @@ public void putTaskConfigs(String connector, Map<ConnectorTaskId, Map<String, St
         // In theory, there is only a single writer and we shouldn't need this lock since the background thread should
         // not invoke any callbacks that would conflict, but in practice this guards against inconsistencies due to
         // the root config being updated.
-        Map<String, Integer> newTaskCounts = new HashMap<>();
+        int newTaskCount = 0;
         synchronized (lock) {
             // Validate tasks in this assignment. Any task configuration updates should include updates for *all* tasks
             // in the connector -- we should have all task IDs 0 - N-1 within a connector if any task is included here
-            Map<String, Set<Integer>> updatedConfigIdsByConnector = taskIdsByConnector(configs);
-            for (Map.Entry<String, Set<Integer>> taskConfigSetEntry : updatedConfigIdsByConnector.entrySet()) {
-                if (!completeTaskIdSet(taskConfigSetEntry.getValue(), taskConfigSetEntry.getValue().size())) {
-                    log.error("Submitted task configuration contain invalid range of task IDs, ignoring this submission");
-                    throw new ConnectException("Error writing task configurations: found some connectors with invalid connectors");
-                }
-                newTaskCounts.put(taskConfigSetEntry.getKey(), taskConfigSetEntry.getValue().size());
+            Set<Integer>  taskConfigSet = taskIds(connector, configs);
+            if (!completeTaskIdSet(taskConfigSet, taskConfigSet.size())) {
```

Comment:
It doesn't seem right to use the size of the same set that we're checking here as the expected size. Maybe we should use `configs.size()` instead?